crates/trie/sparse/src/trie.rs

```diff
@@ -409,7 +409,7 @@ impl<P> RevealedSparseTrie<P> {
                     self.reveal_node_or_hash(child_path, &ext.child)?;
                 }
             },
-            TrieNode::Leaf(leaf) => match self.nodes.entry(path) {
+            TrieNode::Leaf(leaf) => match self.nodes.entry(path.clone()) {
```

Comment:
to prevent these clones, it might be better to make the keys in these maps (like mentioned in the issue) either `&Nibbles` or `Arc<Nibbles>`